numpyro/distributions/continuous.py

```diff
@@ -651,6 +651,156 @@ def variance(self):
+        # no expand needed here ??
+        return np.sum(np.square(self._unbroadcasted_cov_factor), axis=-1) + self._unbroadcasted_cov_diag
+
+    @lazy_property
+    def scale_tril(self):
+        # The following identity is used to increase the numerically computation stability
+        # for Cholesky decomposition (see http://www.gaussianprocess.org/gpml/, Section 3.4.3):
+        #     W @ W.T + D = D1/2 @ (I + D-1/2 @ W @ W.T @ D-1/2) @ D1/2
+        # The matrix "I + D-1/2 @ W @ W.T @ D-1/2" has eigenvalues bounded from below by 1,
+        # hence it is well-conditioned and safe to take Cholesky decomposition.
+        cov_diag_sqrt_unsqueeze = np.expand_dims(np.sqrt(self._unbroadcasted_cov_diag), axis=-1)
+        Dinvsqrt_W = self._unbroadcasted_cov_factor / cov_diag_sqrt_unsqueeze
+        K = np.matmul(Dinvsqrt_W, np.transpose(Dinvsqrt_W, axes=[-1, -2]))
+        K = np.add(K, np.identity(K.shape(-1)))
+        scale_tril = cov_diag_sqrt_unsqueeze * np.linalg.cholesky(K)
+        # again no expand function here ?
+        return scale_tril
+
+    @lazy_property
+    def covariance_matrix(self):
+        covariance_matrix = self._unbroadcasted_cov_diag + np.matmul(self._unbroadcasted_cov_factor,
+                                    np.transpose(self._unbroadcasted_cov_factor, axes=[-1, -2]))
+        return covariance_matrix
+
+    @lazy_property
+    def precision_matrix(self):
+        # We use "Woodbury matrix identity" to take advantage of low rank form::
+        #     inv(W @ W.T + D) = inv(D) - inv(D) @ W @ inv(C) @ W.T @ inv(D)
+        # where :math:`C` is the capacitance matrix.
+        Wt_Dinv = (np.transpose(self._unbroadcasted_cov_factor, axes=[-1, -2])
+                   / np.expand_dims(self._unbroadcasted_cov_diag, axis=-2))
+        A = solve_triangular(Wt_Dinv, self._capacitance_tril, lower=True)
+        return np.reciprocal(self._unbroadcasted_cov_diag) - np.matmul(np.transpose(A, axes=[-1, -2]), A)
+
+    def sample(self, key, sample_shape=()):
+        W_shape = sample_shape + self.cov_factor.shape[-1:]
+        eps_W = random.normal(key, W_shape)
+        eps_D = random.normal(key, sample_shape)
+        return (self.loc + _batch_mv(self._unbroadcasted_cov_factor, eps_W)
+                + np.sqrt(self._unbroadcasted_cov_diag) * eps_D)
+
+    def log_prob(self, value):
+        if self._validate_args:
+            self._validate_sample(value)
+        diff = value - self.loc
+        M = _batch_lowrank_mahalanobis(self._unbroadcasted_cov_factor,
+                                       self._unbroadcasted_cov_diag,
+                                       diff,
+                                       self._capacitance_tril)
+        log_det = _batch_lowrank_logdet(self._unbroadcasted_cov_factor,
+                                        self._unbroadcasted_cov_diag,
+                                        self._capacitance_tril)
+        return -0.5 * (self._event_shape[0] * np.log(2 * np.pi) + log_det + M)
```

Comment:
nit: it is better to use `self.loc.shape[-1]` for event size. We don't use `self._event_shape` in NumPyro.